jmeter: 3.3 -> 4.0

markus1189 opened the pull request:
[x] Tested execution of binary file

markus1189 committed:
jmeter: 3.3 -> 4.0

volth commented:
I am not sure that ```$ARGS``` interpolated correctly in lines [13-14](https://github.com/markus1189/nixpkgs/blob/a48a376f36833817c5d99ffc37b9094e061deed1/pkgs/applications/networking/jmeter/default.nix#L13)

It looks like it is interpolated to empty string

markus1189 commented:
I think you're right @volth, I am currently investigating it seems like there is nothing to substitute anyway because the script does already in 3.3 no longer have this line.

markus1189 commented:
@volth Okay I refactored the package a bit, could you make another review pass?

Things done:

1) extracted `version` attribute
2) added test in checkPhase whether jmeter runs (I had to reorder the phases, true?)
3) prefixed some of the `bin/` scripts with the `jmeter-` prefix
4) wrapped the jmeter program because 4.0 checks JAVA_HOME anyway

fpletz commented on pkgs/applications/networking/jmeter/default.nix:
```diff
@@ -1,19 +1,37 @@
-  buildInputs = [ jre ];
+  buildInputs = [ jre makeWrapper ];
+
+  phases = [ "unpackPhase" "installPhase" "checkPhase" "fixupPhase" ];
```
Please use `dontBuild = true` instead.

markus1189 replied:
I also had to change the order of the `checkPhase` though

fpletz commented on pkgs/applications/networking/jmeter/default.nix:
```diff
@@ -1,19 +1,37 @@
   };
 
-  buildInputs = [ jre ];
+  buildInputs = [ jre makeWrapper ];
```
`makeWrapper` should be in `nativeBuildInputs`.

volth commented:
As I remember it should be safe to skip ```buildPhase``` without declaring its absence in ```phases``` or ```doNotBuild```

markus1189 commented:
@volhovm Do you refer to the `dontBuild = true` comment?

I can add that, but I still need that line because I am reordering the `checkPhase` to be before the `installPhase`.  Otherwise the check does not work.  Should I instead perform the `wrapProgram` in some other phase?

volth commented:
@markus1189 this: https://github.com/markus1189/nixpkgs/pull/2/files

markus1189 commented:
@volth Already included in the latest commit.  Thanks for the pr that is much ncier

Mic92 commented:
there is a bit more wrapping needed:
```console
$ /nix/store/34wdgmwq6gqhzbyzx18sig78jh4yrygz-jmeter-4.0/bin/jmeter-create-rmi-keystore.sh: line 18: keytool: command not found
Copy the generated rmi_keystore.jks to jmeter/bin folder or reference it in property 'server.rmi.ssl.keystore.file'
$ /nix/store/34wdgmwq6gqhzbyzx18sig78jh4yrygz-jmeter-4.0/bin/jmeter-heapdump.sh: line 24: java: command not found
$ /nix/store/34wdgmwq6gqhzbyzx18sig78jh4yrygz-jmeter-4.0/bin/jmeter-mirror-server.sh: line 27: java: command not found
$ /nix/store/34wdgmwq6gqhzbyzx18sig78jh4yrygz-jmeter-4.0/bin/jmeter-mirror-server.sh: line 27: java: command not found
```

markus1189 commented:
Done ;)

I really like how this came out, it's a good improvement of the old package, thanks for all the feedback!

Mic92 commented:
the following are still missing java:

```console
$ /nix/store/6867z1rhini3q3qwps2d8brd2k49r2xs-jmeter-4.0/bin/jmeter-shutdown.sh: line 24: java: command not found
$ /nix/store/6867z1rhini3q3qwps2d8brd2k49r2xs-jmeter-4.0/bin/jmeter-stoptest.sh: line 24: java: command not found
```

markus1189 commented:
Updated and added more tests.

Is the use of `timeout` okay for the test of the jmeter-mirror-server?

Mic92 commented:
The test still has `java` in PATH, if you run it outside, it still misses it.

markus1189 commented:
Should be fixed now

Mic92 committed:
jmeter: nitpicks
